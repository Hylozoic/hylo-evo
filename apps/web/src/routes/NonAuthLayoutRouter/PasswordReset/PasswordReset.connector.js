import { connect } from 'react-redux'
import sendPasswordReset from 'store/actions/sendPasswordReset'

export const mapDispatchToProps = {
  sendPasswordReset
}

export default connect(null, mapDispatchToProps)
