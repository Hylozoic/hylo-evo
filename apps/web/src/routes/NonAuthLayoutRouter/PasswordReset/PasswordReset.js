import React, { useState, useEffect, useRef } from 'react'
import { useTranslation } from 'react-i18next'
import validator from 'validator'
import TextInput from 'components/TextInput'
import Button from 'components/Button'
import classes from './PasswordReset.module.scss'

function PasswordReset ({ className, sendPasswordReset }) {
  const [email, setEmail] = useState('')
  const [success, setSuccess] = useState(false)
  const [error, setError] = useState(false)
  const emailRef = useRef(null)
  const { t } = useTranslation()

  useEffect(() => {
    emailRef.current.focus()
  }, [])

  const submit = () => {
    sendPasswordReset(email)
      .then(({ error }) => {
        if (error) {
          setError(true)
          setSuccess(false)
        } else {
          setSuccess(true)
          setError(false)
        }
      })
  }

  const onChange = event => {
    setEmail(event.target.value)
    setSuccess(false)
    setError(false)
  }

  const canSubmit = validator.isEmail(email)

  return (
    <div className={className}>
      <div className={classes.formWrapper}>
        <h1 className={classes.title}>{t('Reset Your Password')}</h1>
        <div className={classes.subtitle}>
          {t("Enter your email address and we'll send you an email that lets you reset your password.")}
        </div>
        {success && <div className={classes.success}>{t('If your email address matched an account in our system, we sent you an email. Please check your inbox.')}</div>}
        {error && <div className={classes.error}>{t('There was a problem with your request. Please check your email and try again.')}</div>}

        <label>{t('Your email address')}</label>
        <TextInput
          autoFocus
          inputRef={emailRef}
          name='email'
          noClearButton
          onChange={onChange}
          onEnter={submit}
          className={classes.field}
          type='text'
          value={email}
        />

        <Button
          className={classes.submit}
          label={t('Reset')}
          color={canSubmit ? 'green' : 'gray'}
          onClick={canSubmit ? submit : null}
        />
      </div>
    </div>
  )
}

export default PasswordReset
