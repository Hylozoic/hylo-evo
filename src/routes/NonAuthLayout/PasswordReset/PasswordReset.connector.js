import { connect } from 'react-redux'
import { resetPassword } from './PasswordReset.store.js'

export const mapDispatchToProps = {
  resetPassword
}

export default connect(null, mapDispatchToProps)
