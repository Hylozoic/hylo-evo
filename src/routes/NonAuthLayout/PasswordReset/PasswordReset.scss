.title {
  composes: title from '../Login/Login.scss'
}

.subtitle {
  composes: caption-lt-lg from 'css/typography.scss';
  margin-bottom: 25px;
}

.success {
  composes: caption-lt-lg from 'css/typography.scss';
  border: 1px solid $color-caribbean-green;
  color: $color-caribbean-green;
  padding: 5px;
  margin-bottom: 25px;
}

.error {
  composes: caption-lt-lg from 'css/typography.scss';
  border: 1px solid $color-amaranth;
  color: $color-amaranth;
  padding: 5px;
  margin-bottom: 25px;
}

.field {
  composes: field from '../Login/Login.scss'
}

.field-label {
  composes: field-label from '../Login/Login.scss'
}

.submit {
  composes: submit from '../Login/Login.scss'
}
