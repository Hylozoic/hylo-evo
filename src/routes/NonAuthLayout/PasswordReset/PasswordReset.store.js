export const MODULE_NAME = 'PasswordReset'
export const RESET_PASSWORD = `${MODULE_NAME}/RESET_PASSWORD`

export function resetPassword (email) {
  return {
    type: RESET_PASSWORD,
    payload: {
      api: {method: 'post', path: '/noo/user/password', params: {email, evo: true}}
    }
  }
}
