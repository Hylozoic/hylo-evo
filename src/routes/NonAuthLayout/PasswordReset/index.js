import component from './PasswordReset'
import connector from './PasswordReset.connector'
export default connector(component)
