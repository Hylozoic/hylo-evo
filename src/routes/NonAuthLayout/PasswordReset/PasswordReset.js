import React, { Component } from 'react'
import validator from 'validator'
import TextInput from 'components/TextInput'
import Button from 'components/Button'
import './PasswordReset.scss'

export default class PasswordReset extends Component {
  constructor (props) {
    super(props)
    this.state = {email: '', success: false, error: false}
  }

  componentDidMount () {
    this.email.focus()
  }

  submit () {
    const { email } = this.state
    this.props.resetPassword(email)
    .then(({ error }) => {
      if (error) {
        this.setState({error})
      } else {
        this.setState({success: true})
      }
    })
  }

  render () {
    const { className } = this.props
    const onChange = event => this.setState({
      email: event.target.value,
      success: false
    })
    const { email, success, error } = this.state

    const canSubmit = validator.isEmail(email)

    return <div className={className}>
      <h1 styleName='title'>Reset Your Password</h1>
      <div styleName='subtitle'>
        Enter your email address and we'll send you an email that lets you reset your password.
      </div>
      {success && <div styleName='success'>If your email address matched an account in our system, we sent you an email. Please check your inbox.</div>}
      {error && <div styleName='error'>There was a problem with your request. Please check your email and try again.</div>}
      <div styleName='field'>
        <label styleName='field-label'>Your email address</label>
        <TextInput type='text' name='email' onChange={onChange} value={email}
          inputRef={input => { this.email = input }} noClearButton />
      </div>
      <Button styleName='submit' label='Reset' color={canSubmit ? 'green' : 'gray'}
        onClick={canSubmit ? () => this.submit() : null} />
    </div>
  }
}
