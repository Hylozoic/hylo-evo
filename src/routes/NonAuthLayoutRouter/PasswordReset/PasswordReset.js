import React, { Component } from 'react'
import { withTranslation } from 'react-i18next'
import validator from 'validator'
import TextInput from 'components/TextInput'
import Button from 'components/Button'
import './PasswordReset.scss'

class PasswordReset extends Component {
  constructor (props) {
    super(props)
    this.state = { email: '', success: false, error: false }
  }

  componentDidMount () {
    this.email.focus()
  }

  submit () {
    const { email } = this.state
    this.props.sendPasswordReset(email)
      .then(({ error }) => {
        if (error) {
          this.setState({ error })
        } else {
          this.setState({ success: true })
        }
      })
  }

  render () {
    const { className, t } = this.props
    const onChange = event => this.setState({
      email: event.target.value,
      success: false
    })
    const { email, success, error } = this.state

    const canSubmit = validator.isEmail(email)

    return (
      <div className={className}>
        <div styleName='formWrapper'>
          <h1 styleName='title'>{this.props.t('Reset Your Password')}</h1>
          <div styleName='subtitle'>
            {this.props.t("Enter your email address and we'll send you an email that lets you reset your password.")}
          </div>
          {success && <div styleName='success'>{this.props.t('If your email address matched an account in our system, we sent you an email. Please check your inbox.')}</div>}
          {error && <div styleName='error'>{this.props.t('There was a problem with your request. Please check your email and try again.')}</div>}

          <label>{this.props.t('Your email address')}</label>
          <TextInput
            autoFocus
            inputRef={input => { this.email = input }}
            name='email'
            noClearButton
            onChange={onChange}
            onEnter={() => { this.submit() }}
            styleName='field'
            type='text'
            value={email}
          />

          <Button
            styleName='submit' label={t('Reset')} color={canSubmit ? 'green' : 'gray'}
            onClick={canSubmit ? () => this.submit() : null}
          />
        </div>
      </div>
    )
  }
}
export default withTranslation()(PasswordReset)
