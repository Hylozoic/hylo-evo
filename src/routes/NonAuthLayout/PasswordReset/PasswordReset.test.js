import React from 'react'
import { shallow } from 'enzyme'
import PasswordReset from './PasswordReset'

describe('Signup', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<PasswordReset />)
    expect(wrapper).toMatchSnapshot()
  })
})
