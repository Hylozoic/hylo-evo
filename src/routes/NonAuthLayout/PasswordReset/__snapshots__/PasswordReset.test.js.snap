// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup renders correctly 1`] = `
<div>
  <h1
    data-stylename="title"
  >
    Reset Your Password
  </h1>
  <div
    data-stylename="subtitle"
  >
    Enter your email address and we'll send you an email that lets you reset your password.
  </div>
  <div
    data-stylename="field"
  >
    <label
      data-stylename="field-label"
    >
      Your email address
    </label>
    <TextInput
      inputRef={[Function]}
      name="email"
      noClearButton={true}
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <Button
    color="gray"
    data-stylename="submit"
    label="Reset"
    onClick={null}
  />
</div>
`;
